UefiTestingPkg/FunctionalSystemTests/MemoryProtectionTest/App/MemoryProtectionTestApp.c

```diff
@@ -861,14 +890,28 @@ UefiPageGuard (
   MEMORY_PROTECTION_TEST_CONTEXT  MemoryProtectionContext = (*(MEMORY_PROTECTION_TEST_CONTEXT *)Context);
   EFI_PHYSICAL_ADDRESS            ptr;
   EFI_STATUS                      Status;
+  UINT64                          Attributes;
 
   DEBUG ((DEBUG_INFO, "%a - Testing Type: %a\n", __FUNCTION__, MEMORY_TYPES[MemoryProtectionContext.TargetMemoryType]));
 
-  if (MemoryProtectionContext.DynamicActive) {
-    if (mNonstopModeProtocol == NULL) {
-      UT_ASSERT_NOT_EFI_ERROR (GetNonstopProtocol ());
+  if (MemoryProtectionContext.TestingMethod == MemoryProtectionTestMemoryAttributeProtocol) {

```

Comment:
This function is getting long. Can you split up some of the conditional blocks into separate functions that get called from here?

Reply:
I'd prefer not to split things up unless this is a sticking point. I split up the entry point function because that was a few hundred lines. This one is only 100 which seems pretty normal to me.

Reply:
We decided upon a 60 line max ([contributing.md](https://github.com/microsoft/mu_plus/blob/release/202208/CONTRIBUTING.md#implementation-limits)) but that is almost never enforced.

It's not so much the total length (at this point in time) but the fact that the function is primarily composed of exclusive blocks of functionality. It would be easier to follow and extend if broken into separate functions that perform separate functional repsonsibilities. But, it's more of a nice to have so feel free to pass.

Reply:
Looking through the file more, you'd have to do this in most functions exercising a memory protection. There's not a need for that level of change to do this.